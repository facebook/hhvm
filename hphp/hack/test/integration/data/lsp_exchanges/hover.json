[
{
  "jsonrpc": "2.0",
  "id": 1,
  "method": "initialize",
  "params":
    {
      "initializationOptions":{},
      "processId":null,
      "rootPath":"${root_path}",
      "capabilities":{}
    }
},
{
  "jsonrpc": "2.0",
  "method": "textDocument/didOpen",
  "params": {
     "textDocument": {
       "uri":"${php_file_uri}",
       "languageId": "hack",
       "version":1,
       "text":"${php_file}"
     }
   }
},
{
  "comment": "hover over function invocation",
  "jsonrpc": "2.0",
  "method": "textDocument/hover",
  "id": 2,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
      },
      "position": {"line": 3, "character": 16}
   }
},
{
  "comment": "hover over whitespace",
  "jsonrpc": "2.0",
  "method": "textDocument/hover",
  "id": 3,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
      },
      "position": {"line": 3, "character": 1}
   }
},
{
  "comment": "hover over a keyword",
  "jsonrpc": "2.0",
  "method": "textDocument/hover",
  "id": 4,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
      },
      "position": {"line": 2, "character": 1}
   }
},
{
  "comment": "hover over a comment",
  "jsonrpc": "2.0",
  "method": "textDocument/hover",
  "id": 5,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
      },
      "position": {"line": 1, "character": 4}
   }
},
{
  "comment": "hover past the end of a line",
  "jsonrpc": "2.0",
  "method": "textDocument/hover",
  "id": 6,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
      },
      "position": {"line": 3, "character": 100}
   }
},
{
  "comment": "hover past the end of a file",
  "jsonrpc": "2.0",
  "method": "textDocument/hover",
  "id": 7,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
      },
      "position": {"line": 300, "character": 0}
   }
},
{
  "jsonrpc": "2.0",
  "id": 8,
  "method": "shutdown",
  "params": {}
}
]
