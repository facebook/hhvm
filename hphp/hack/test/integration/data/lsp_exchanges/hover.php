<?hh  //strict
// comment
function a_hover(): int {
  return b_hover();
}

function b_hover(): int {
  return 42;
}
