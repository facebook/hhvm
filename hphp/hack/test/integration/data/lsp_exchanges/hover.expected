[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
      "capabilities": {
        "textDocumentSync": {
          "openClose": true,
          "change": 2,
          "willSave": false,
          "willSaveWaitUntil": false,
          "save": {
            "includeText": false
          }
        },
        "hoverProvider": true,
        "completionProvider": {
          "resolveProvider": true,
          "triggerCharacters": [
            "$",
            ">",
            "\\",
            ":",
            "<",
            "[",
            "'",
            "\""
          ]
        },
        "signatureHelpProvider": {
          "triggerCharacters": [
            "(",
            ","
          ]
        },
        "definitionProvider": true,
        "referencesProvider": true,
        "documentHighlightProvider": true,
        "documentSymbolProvider": true,
        "workspaceSymbolProvider": true,
        "codeActionProvider": false,
        "documentFormattingProvider": true,
        "documentRangeFormattingProvider": true,
        "documentOnTypeFormattingProvider": {
          "firstTriggerCharacter": ";",
          "moreTriggerCharacter": [
            "}"
          ]
        },
        "renameProvider": true,
        "typeCoverageProvider": true,
        "rageProvider": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 2,
    "result": {
      "contents": [
        {
          "language": "hack",
          "value": "int"
        }
      ],
      "range": {
        "start": {
          "line": 3,
          "character": 9
        },
        "end": {
          "line": 3,
          "character": 16
        }
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 3,
    "result": null
  },
  {
    "jsonrpc": "2.0",
    "id": 4,
    "result": null
  },
  {
    "jsonrpc": "2.0",
    "id": 5,
    "result": null
  },
  {
    "jsonrpc": "2.0",
    "id": 6,
    "result": null
  },
  {
    "jsonrpc": "2.0",
    "id": 7,
    "result": null
  },
  {
    "jsonrpc": "2.0",
    "id": 8,
    "result": null
  }
]